### PR TITLE
Refactor and extend Playwright tests

### LINKED ISSUE #2020: [ENH] - Add conda-store-ui playwright test
### Feature description

We need to add a test that opens the jlab conda-store-ui, goes through the login process and makes sure that environments can be created and builds deleted. 

### Value and/or benefit

We catch failures early rather than later. Less burden on the person testing for release. 

### Anything else?

_No response_

### LINKED ISSUE #2642: [BUG] - TimeoutError in Playwright Notebook tests
### Describe the bug

We are encountering a `TimeoutError` when running the `test_notebook` test within our CI pipeline in mostly of our current PRs. The error occurs during the `reset_workspace` method in the [navigator.py](https://github.com/nebari-dev/nebari/blob/develop/tests/common/run_notebook.py) file, specifically when waiting for the page load state to reach "networkidle." The test fails with the following traceback:

```plaintext
  File "/home/runnerx/actions-runner/_work/nebari/nebari/tests/common/navigator.py", line 252, in reset_workspace
    popup = self._check_for_kernel_popup()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/actions-runner/_work/nebari/nebari/tests/common/navigator.py", line 236, in _check_for_kernel_popup
    self.page.wait_for_load_state("networkidle")
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/playwright/sync_api/_generated.py", line 8949, in wait_for_load_state
    self._sync(self._impl_obj.wait_for_load_state(state=state, timeout=timeout))
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/playwright/_impl/_sync_base.py", line 115, in _sync
    return task.result()
           ^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/playwright/_impl/_page.py", line 540, in wait_for_load_state
    return await self._main_frame.wait_for_load_state(**locals_to_params(locals()))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/playwright/_impl/_frame.py", line 243, in wait_for_load_state
    return await self._wait_for_load_state_impl(state, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runnerx/miniconda3/envs/test/lib/python3.11/site-packages/playwright/_impl/_frame.py", line 271, in _wait_for_load_state_impl
    await waiter.result()
playwright._impl._errors.TimeoutError: Timeout 30000ms exceeded.
```

During an initial debug of this issue, it was observed that waiting for the "networkidle" state was not required. A simple `wait_for_load_state` call without specifying the state resolved the timeout issue. However, additional issues were encountered related to environment selection within the kernel selector popups in the tests themselves, which might require further investigation and refactoring.

### Expected behavior

- Playwright tests succeeding 

### OS and architecture in which you are running Nebari

Linux

### How to Reproduce the problem?

The test that triggers this error is located in the `test_playwright.py` file under the `test_notebook` function, which relies on the following setup:

```python
from tests.common.playwright_fixtures import navigator_parameterized
from tests.common.run_notebook import Notebook

@navigator_parameterized(instance_name="small-instance")
def test_notebook(navigator, test_data_root):
    test_app = Notebook(navigator=navigator)
    notebook_name = "test_notebook_output.ipynb"
    notebook_path = test_data_root / notebook_name
    assert notebook_path.exists()
    with open(notebook_path, "r") as notebook:
        test_app.nav.write_file(filepath=notebook_name, content=notebook.read())
    test_app.run(
        path=notebook_name,
        expected_outputs=["success: 6"],
        conda_env="default *",
        timeout=500,
    )
```

You can follow the [readme](https://github.com/nebari-dev/nebari/blob/develop/tests/tests_e2e/playwright/README.md) instructions on how to run the e2e tests locally. 

### Command output

_No response_

### Versions and dependencies used.

_No response_

### Compute environment

None

### Integrations

_No response_

### Anything else?

This issue must be resolved as it is a significant obstacle in our testing workflows, causing disruptions and preventing other tests from running successfully in most of our recent PRs.

### PR DESCRIPTION
**Pull request by viniciusdc**
## Reference Issues or PRs

related to #2020 
closes #2642 

## What does this implement/fix?

- Refactor playwright main navigator class to allow better flexibility when extending test suit;
- Add new conda-store UI class to test its accessibility and running;
- Rewrote some internal conditional logic when locating objects within the jupyterlab workspace;
- Add the capability to name video artifacts based on the executed test name.

_Put a `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Any other comments?

I will open a follow-up PR to extend the conda-store test suit based on @kcpevey suggestion https://github.com/nebari-dev/nebari/issues/2020#issuecomment-1768918050

### CODE REVIEW

**Comment by viniciusdc** [comment]
Tests are failing due to the problem outlined in #2643, but we can still validate the playwright tests are now working:

https://github.com/user-attachments/assets/d2dce6dc-b4ec-4320-8230-27be64b487b4


https://github.com/user-attachments/assets/fbdcddcb-edbc-48c9-9d73-742d4c0ee4ea


https://github.com/user-attachments/assets/a03e8a8d-6e4a-40a4-87fe-10b8d8b8e1b1


https://github.com/user-attachments/assets/ab812e10-5b8d-4e1b-b980-1530b02a6c2a

**Review by marcelovilla — APPROVED** [review]
Awesome work @viniciusdc! I think this is a good first step in refactoring our tests and making them more extensible.

Regarding issue #2020, I'd say this PR does not necessarily closes it as based on the description, we also want to test environment creation and build deletion. I'm not saying we need to address that right away in this PR and I'll let you decide whether it should be closed or not.

File: .github/workflows/test_local_integration.yaml [hunk 1]
```diff
@@ -172,7 +172,7 @@ jobs:
           # create environment file
           envsubst < .env.tpl > .env
           # run playwright pytest tests in headed mode with the chromium browser
-          xvfb-run pytest --browser chromium
+          xvfb-run pytest --browser chromium --slowmo 300 --headed
```
[the remark this answers is not in the document]
Comment (viniciusdc) [reply .github/workflows/test_local_integration.yaml:175]: The only downside would be that all related playwright actions within a test (click a button, query, refresh etc) will take a slight increase in execution time which in return increases the time for the overall playwright pytest suit to complete -- though, I advice that this is essential for some components to load properly

**Comment by viniciusdc** [comment]
> 'd say this PR does not necessarily closes it as based on the description, we also want to test environment creation and build deletion.
@marcelovilla, good point; I will remove that closing tag and add a simple ref. Instead, I saw a note from @kcpevey on recent playwright tests added to conda-store; I was going to open a follow-up issue to extend this base class, as the main goal for this one was to at least catch the problem with starting the plugin at all.

**Pull request merged**